dsm/common.go

```diff
@@ -263,4 +264,91 @@ func undoTFstate(param_type string, d *schema.ResourceData) diag.Diagnostics {
 		  Detail:   fmt.Sprintf("[E]: API: PATCH crypto/v1/keys: %s cannot change on update. Please retain it to old value: %s -> %s", param_type, old_val, new_val),
 	})
 	return diags
-}
\ No newline at end of file
+}
+
+// expiry_date create and update
+func parseTimeToDSM(expiry_date string) (string, diag.Diagnostics){
+	layoutRFC := "2006-01-02T15:04:05Z"
+	layoutDSM := "20060102T150405Z"
+	ddate, newerr := time.Parse(layoutRFC, expiry_date)
+	if newerr != nil {
+		return "", diag.FromErr(newerr)
+	}
+	return ddate.Format(layoutDSM), nil
+}
+
+/* Set the key_ops in tf state.
+This function is required, request key_ops order and response key_ops order mighet differ.
+Hence, during `terraform plan`, if there are no changes in key_ops, it should not show any changes.
+*/
```

Comment:
Also we should be consistent with our doc comments on these functions. If you refer to the section on comments for Go ( [here](https://go.dev/doc/comment) ), they show using the regular `//` for functions.

Reply:
Done. Modified all such related comments.